packages/core/content-manager/shared/contracts/collection-types.ts

```diff
@@ -1,12 +1,18 @@
 import { errors } from '@strapi/utils';
-import { Schema, Common, EntityService } from '@strapi/types';
+import { Schema, Common, Documents } from '@strapi/types';
 
-// Admin entity response follows the same format as the entity service
-type Entity = EntityService.Result<Common.UID.Schema>;
-type PaginatedEntities = EntityService.PaginatedResult<Common.UID.Schema>;
+type PaginatedDocuments = Documents.PaginatedResult<Common.UID.Schema>;
+type PaginationQuery = Documents.Params.Pagination.PageNotation;
+type SortQuery = Documents.Params.Sort.StringNotation<Common.UID.Schema> & string;
 
-type PaginationQuery = EntityService.Params.Pagination.PageNotation;
-type SortQuery = EntityService.Params.Sort.StringNotation<Common.UID.Schema> & string;
+// Admin document response follows the same format as the document service
+type Document = Documents.Result<Common.UID.Schema>;
+type DocumentMetadata = {
+  // All status of the returned locale
+  availableStatus: Document[];
+  // Available locales within the same status of the returned document
+  availableLocales: Document[];
+};
```

Comment:
@joshuaellis we originally agreed to another metadata format:

```
type Response {
  data: Document
  meta?: {
	publicationState: PublicationState
	entries: [
		{ locale: Locale, updatedAt, publicationState: PublicationState },
	]
  };
}
```

While looking at it again, I find the following issues with it:
- It's hard to understand. 
   - What is publication state? the publication state of the returned document?  Why is it returned inside meta for the ressponse, but inside the list of entries in metadata?
   - What are entries? People might be confused by the term, and internally we are just not mentioning what is an entry at all (we use "versions" to specify different variants of the same document.
   
- It's not scalable.
  - Returning all locales and all its publication states is not scalable neither useful. Rather, I think its more useful to know if there is a publsihed version of the returned locale, and all of the other draft locales available.
  
  wdyt?

Reply:
My current proposal would need less versions to be loaded, and I find it a bit easier to understand.

In a glance you can find if the document is published, and also find its other locales.

Reply:
As discussed, lets return only the necessary parts of documents.
```
availableStatus: Pick<Document, 'id' | AT_FIELDS>[];
availableLocales: Pick<Document, 'id' | AT_FIELDS | 'locale'>
```